templates/company-board/contracts/CompanyBoardTemplate.sol

```diff
@@ -172,17 +184,22 @@ contract CompanyBoardTemplate is BaseTemplate {
         c.shareToken = address(_shareToken);
     }
 
-    function _getCache(address _owner) internal view returns (Kernel dao, MiniMeToken boardToken, MiniMeToken shareToken) {
+    function _popDaoCache(address _owner) internal returns (Kernel) {
         Cache storage c = cache[_owner];
-        require(c.dao != address(0) && c.boardToken != address(0) && c.shareToken != address(0), ERROR_MISSING_DAO_CACHE);
+        require(c.dao != address(0), ERROR_MISSING_CACHE);
 
-        dao = Kernel(c.dao);
-        boardToken = MiniMeToken(c.boardToken);
-        shareToken = MiniMeToken(c.shareToken);
+        Kernel dao = Kernel(c.dao);
+        delete c.dao;
+        return dao;
     }
 
-    function _popCache(address _owner) internal returns (Kernel dao, MiniMeToken boardToken, MiniMeToken shareToken) {
-        (dao, boardToken, shareToken) = _getCache(_owner);
-        delete cache[_owner];
+    function _popTokensCache(address _owner) internal returns (MiniMeToken boardToken, MiniMeToken shareToken) {
+        Cache storage c = cache[_owner];
+        require(c.boardToken != address(0) && c.shareToken != address(0), ERROR_MISSING_CACHE);
+
+        boardToken = MiniMeToken(c.boardToken);
+        shareToken = MiniMeToken(c.shareToken);
+        delete c.boardToken;
+        delete c.shareToken;
     }
```

Comment:
In the same vein of rolling things up, I liked your suggestion of just popping everything at the start of `setupInstance()`.

Reply:
same argument here, `stack too deep` issues